src/compas_timber/elements/plate.py

```diff
@@ -0,0 +1,242 @@
+        """Compute the geometry of the element.
+
+        Parameters
+        ----------
+        include_features : bool, optional
+            If ``True``, include the features in the computed geometry.
+            If ``False``, return only the base geometry.
+
+        Returns
+        -------
+        :class:`compas.datastructures.Mesh` | :class:`compas.geometry.Brep`
+
+        """
+        plate_geo = self.shape
+        if include_features:
+            for feature in self.features:
+                try:
+                    plate_geo = feature.apply(plate_geo)
+                except FeatureApplicationError as error:
+                    self.debug_info.append(error)
+        return plate_geo
+
+    def compute_aabb(self, inflate=0.0):
+        # type: (float) -> compas.geometry.Box
+        """Computes the Axis Aligned Bounding Box (AABB) of the element.
+
+        Parameters
+        ----------
+        inflate : float, optional
+            Offset of box to avoid floating point errors.
+
+        Returns
+        -------
+        :class:`~compas.geometry.Box`
+            The AABB of the element.
+
+        """
+        vertices = [point for point in self.outline.points]
+        for point in self.outline.points:
+            vertices.append(point + self.vector)
+        box = Box.from_points(vertices)
+        box.xsize += inflate
+        box.ysize += inflate
+        box.zsize += inflate
+        return box
+
+    def compute_obb(self, inflate=0.0):
+        # type: (float | None) -> compas.geometry.Box
+        """Computes the Oriented Bounding Box (OBB) of the element.
+
+        Parameters
+        ----------
+        inflate : float
+            Offset of box to avoid floating point errors.
+
+        Returns
+        -------
+        :class:`compas.geometry.Box`
+            The OBB of the element.
+
+        """
+        vertices = [point for point in self.outline.points]
+        for point in self.outline.points:
+            vertices.append(point + self.vector)
+        for point in vertices:
+            point.transform(Transformation.from_change_of_basis(Frame.worldXY(), self.frame))
+        obb = Box.from_points(vertices)
+        obb.xsize += inflate
+        obb.ysize += inflate
+        obb.zsize += inflate
+
+        obb.transform(Transformation.from_change_of_basis(self.frame, Frame.worldXY()))
+
+        return obb
+
+    def compute_collision_mesh(self):
+        # type: () -> compas.datastructures.Mesh
+        """Computes the collision geometry of the element.
+
+        Returns
+        -------
+        :class:`compas.datastructures.Mesh`
+            The collision geometry of the element.
+
+        """
+        return self.compute_aabb.to_mesh()
+
+    # ==========================================================================
+    # Alternative constructors
+    # ==========================================================================
```

Comment:
move down or remove this section header as there are currently no alternative constructors